dnf5/commands/check-upgrade/check-upgrade.cpp

```diff
@@ -0,0 +1,191 @@
+    auto * arg_parser_this_cmd = get_argument_parser_command();
+    arg_parser_parent_cmd->register_command(arg_parser_this_cmd);
+    arg_parser_parent_cmd->get_group("software_management_commands").register_argument(arg_parser_this_cmd);
+}
+
+void CheckUpgradeCommand::set_argument_parser() {
+    auto & ctx = get_context();
+    auto & parser = ctx.get_argument_parser();
+
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_description("Check for available package upgrades");
+
+    changelogs = dynamic_cast<libdnf::OptionBool *>(
+        parser.add_init_value(std::unique_ptr<libdnf::OptionBool>(new libdnf::OptionBool(false))));
+    auto changelogs_opt = parser.add_new_named_arg("changelogs");
+    changelogs_opt->set_long_name("changelogs");
+    changelogs_opt->set_description("Show changelogs before update.");
+    changelogs_opt->set_const_value("true");
+    changelogs_opt->link_value(changelogs);
+    cmd.register_named_arg(changelogs_opt);
+
+    auto keys = parser.add_new_positional_arg("specs", ArgumentParser::PositionalArg::UNLIMITED, nullptr, nullptr);
+    keys->set_description("List of package specs to check for upgrades");
+    keys->set_parse_hook_func(
+        [this]([[maybe_unused]] ArgumentParser::PositionalArg * arg, int argc, const char * const argv[]) {
+            for (int i = 0; i < argc; ++i) {
+                pkg_specs.emplace_back(argv[i]);
+            }
+            return true;
+        });
+    keys->set_complete_hook_func([&ctx](const char * arg) { return match_specs(ctx, arg, true, false, true, false); });
+    cmd.register_positional_arg(keys);
+
+    advisory_name = std::make_unique<AdvisoryOption>(*this);
+    advisory_security = std::make_unique<SecurityOption>(*this);
+    advisory_bugfix = std::make_unique<BugfixOption>(*this);
+    advisory_enhancement = std::make_unique<EnhancementOption>(*this);
+    advisory_newpackage = std::make_unique<NewpackageOption>(*this);
+    advisory_severity = std::make_unique<AdvisorySeverityOption>(*this);
+    advisory_bz = std::make_unique<BzOption>(*this);
+    advisory_cve = std::make_unique<CveOption>(*this);
+}
+
+void CheckUpgradeCommand::configure() {
+    auto & context = get_context();
+    context.update_repo_metadata_from_specs(pkg_specs);
+    context.set_load_system_repo(true);
+    context.set_load_available_repos(Context::LoadAvailableRepos::ENABLED);
+    if (changelogs->get_value()) {
+        context.base.get_config().get_optional_metadata_types_option().add(libdnf::OPTIONAL_METADATA_TYPES);
+    }
+}
+
+std::unique_ptr<libdnf::cli::output::PackageListSections> CheckUpgradeCommand::create_output() {
+    auto out = std::make_unique<libdnf::cli::output::PackageListSections>();
+    out->setup_cols();
+    return out;
+}
+
+void CheckUpgradeCommand::run() {
+    auto & ctx = get_context();
+    auto & config = ctx.base.get_config();
+
+    libdnf::rpm::PackageQuery full_package_query(ctx.base);
```

Comment:
This is not used anywhere in this function. Could be probably created later inside the `print_changelogs`? So then this parameter would be dropped.